City-of-Helsinki/helsinki-paatokset
# PP-150 & PP258: Meeting decision announcements

Note: This also includes some drush commands changes that don't need to be tested.

**To test:**
- Checkout branch
- Run `drush cr;drush cim -y;drush ap:fs;drush mim ahjo_meetings:latest;drush mim ahjo_decisionmakers:all;`
- Find some meetings that have a decision announcement but no minutes yet: https://helsinki-paatokset.docker.so/fi/admin/content/meetings?title=&field_meeting_id_value=&field_meeting_minutes_published_value=0&field_meeting_agenda_published_value=1&field_meeting_decision_value=1
  - Edit them and set their meeting dates to be in the future
- Find also some that have decision announcements but also have minutes published: https://helsinki-paatokset.docker.so/fi/admin/content/meetings?title=&field_meeting_id_value=&field_meeting_minutes_published_value=1&field_meeting_agenda_published_value=All&field_meeting_decision_value=1
  - Edit some of these and put them in the future too
- View the meetings with decision announcements. They should have the parsed decision contents inside accordions
- View some of the meetings with decision announcements and minutes. They should NOT have the parsed decision contents visible
- Go back to the decision maker pages.
  - Check the documents list and the meeting calendar. The link text should be "Open decision" in the calendar BUT "Agenda" in the documents list.
  - The decisions links should include the #paatostiedote anchor (the motion or minutes links should not) except for the documents list, where the heading link should be without the anchor, but there should be a separate "Read decision announcement" link with the anchor.
- Go to the front page: https://helsinki-paatokset.docker.so/fi/
  - The upcoming meetings with a decision announcement should  also have the label "Open decision" with the anchor
- Go to the meeting calendar link: https://helsinki-paatokset.docker.so/fi/kokouskalenteri
  - Same thing with the decision links
- Go through the code changes. Should anything be refactored? Can you think of cases where the decision announcement handling is missing?
